### DISCUSSION TITLE
Move or remove all dropbox entries

### DISCUSSION THREAD

**Discussion by quanticplus04** [moudey/Shell#491] [Q&A] [answered]
Hi. I managed to move some dropbox entries with modify (find… one by one. Is there a simple way to handle all these entries at once ? Like modify (find=dropbox.process) .-) Depending of menus, Dropbox has many options… I tried shellexview or registry tricks but shell is the safest way without tampering with system
So far one by one entries works. 
//modifier dropbox
//menu  dropbox
menu(title="Dropbox"  type='~Recyclebin'  )
{
}
modify ( find="Transférer copie…|Sauveg.|sur le disque dur|Demander des fichiers…|synchroniser sur" menu="Dropbox")
![image](https://github.com/user-attachments/assets/10246a97-8e4a-49fa-8229-ab389aa04d42)

**Comment by RubicBG** [comment]
- remove all: 
`remove(clsid='{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}' where=this.isuwp)`
- moving one by one is the only option for now:

> STRINGTABLE
LANGUAGE LANG_FRENCH, SUBLANG_FRENCH
{
  210, 	"Partager…"
  211, 	"Partager ce fichier/dossier"
  212, 	"Afficher sur &Dropbox.com"
  213, 	"Consulter ce fichier/dossier sur dropbox.com"
  214, 	"Copier le &lien Dropbox"
  215, 	"Copiez un lien Dropbox vers ce fichier/dossier dans le presse-papiers."
  216, 	"Rendre accessible hors ligne"
  217, 	"Stocker en ligne"
  218, 	"Historique des versions"
  219, 	"Afficher l’historique des modifications de ce fichier/dossier"
  220, 	"Verrouiller"
  221, 	"D\xE9verrouiller"
  222, 	"Synchroniser maintenant"
  223, 	"Obtenir plus d’espace"
  224, 	"D\xE9placer vers Dropbox"
  225, 	"Passer \xE0 Dropbox (Business)"
  226, 	"Transf\xE9rer copie…"
  227, 	"Transf\xE9rer une copie (Business)"
  228, 	"Verrouiller Vault"
  231, 	"Envoyer pour signature"
  232, 	"Sauveg. dans Dropbox…"
  233, 	"Demander des fichiers…"
  234, 	"G\xE9rer la sauvegarde…"
  235, 	"Suspendre la sauvegarde"
  236, 	"Reprendre la sauvegarde"
  237, 	"Ne pas synchroniser sur dropbox.com"
  238, 	"Synchroniser sur dropbox.com"
  239, 	"G\xE9rer l’espace sur le disque dur"
  240, 	"Ne pas synchroniser sur dropbox.com"
  241, 	"Synchroniser sur dropbox.com"
  242, 	"Ne pas synchroniser sur dropbox.com"
  243, 	"Ne pas synchroniser sur dropbox.com"
  244, 	"Cr\xE9ez un dossier et partagez-le"
  245, 	"Renommer"
  247, 	"Aper\xE7u avec Dropbox"
  248, 	"G\xE9rer acc\xE8s fichier"
  249, 	"G\xE9rer acc\xE8s dossier"
}

PS: 
- can you tell me what it shows with
`item(title=str.res(reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32'), -100))`
- are you on any paid plan?

**Reply by quanticplus04** [reply]
TY RubicGB. Trying your solution. Will tell you how it goes. I'm not on a paid plan, I'm retired. I used to be a maintenace technician on computers in supermarckets and I like to code a bit but without any previous teachings, just try and fail scheme. Now I created society and eductional games for kids and families as a hobby. My bro asked me about this dropbox and how to remove all of this junk. I dont't even use dropbox !  I seems like a lot of people have this question and shell is a nice answer. Chatgpt gave me funny answers, lol

**Comment by quanticplus04** [comment]
with
item(title=str.res(reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32'), -100))
it add…"dropbox extension of files…" no change on dropbox labels
![image](https://github.com/user-attachments/assets/dd032bd7-fad6-4244-99f5-ca079f67320f)

**Reply by RubicBG** [reply]
with that I wanted to see if it recognizes a language other than English - it does. But maybe there is an exception...

In short: All commands that can appear in the context menu when Dropbox is installed come from a certain dll file located in the installation directory of the program - for me it is "DropboxExt64.75.0.dll". The name and location may be different for each version and other features such as whether the operating system is 32 or 64 bit. With `str.res(reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32'), -###)` the name of each command can be retrieved, where ### is the code for each command

`modify(menu='Dropbox' where=this.title==str.res(reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32'), -214) tip='Copy a Dropbox link to this file/folder to your clipboard.')`

if the name of the command is changed (which happens) the code will always find it - no need to edit the code again for each change

**Comment by quanticplus04** [comment]
no change with remove all:
remove(clsid='{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}' where=this.isuwp)

![image](https://github.com/user-attachments/assets/15710f86-05b6-4779-a2ea-4fb549ffd1e1)

**Reply by RubicBG** [reply]
https://github.com/moudey/Shell/issues/285#issuecomment-1593408137
> Sometimes this feature does not work for several reasons
> 
>     The key is specific to the system shell32.dll
>     The keys have been loaded into memory and will not be intercepted

try with restarting the PC

**Reply by quanticplus04** [reply]
Restarting pc works…but as soon as I restart explorer it doesn't. I noticed that shell entries without dropbox apears first, then after a while, Dropbox entries.

**Comment by quanticplus04** [comment]
One by one ligne is the solution. Here i moved "Stocker en ligne"
modify ( find="Transférer copie…|Sauveg.|sur le disque dur|Demander des fichiers…|synchroniser sur|Stocker en ligne" menu="Dropbox")
![image](https://github.com/user-attachments/assets/bd39f7d7-1f30-4da5-ab7d-a6f09bd1cdd0)

**Comment by RubicBG** [comment, marked as answer]
Here is another version:
```
$path_db_dll	= reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32')

modify(menu='Dropbox' where=this.title(
		str.res(path_db_dll, -224),
		str.res(path_db_dll, -225),
		str.res(path_db_dll, -232),
		str.res(path_db_dll, -214),
		str.res(path_db_dll, -226),
		str.res(path_db_dll, -227),
		str.res(path_db_dll, -210),
		str.res(path_db_dll, -249),
		str.res(path_db_dll, -248),
		str.res(path_db_dll, -239),
		str.res(path_db_dll, -212),
		str.res(path_db_dll, -231),
		str.res(path_db_dll, -233),
		str.res(path_db_dll, -218),
		str.res(path_db_dll, -216),
		str.res(path_db_dll, -217),
		str.res(path_db_dll, -240),
		str.res(path_db_dll, -241)
	))
```
you can use the same code to remove them all - replace ```modify(menu='Dropbox'``` with ```revove(```

**Reply by quanticplus04** [reply]
Perfect…This way it will work with any language or change of text within the language. Plus it is SPECIFIC to Dropbox as with labels, some others apps could use the same labels.
Ty so much. Nice solutions for those struggling with Dropbox entries
![image](https://github.com/user-attachments/assets/e9cf4f56-5c35-4ba1-99cb-d2cab05ddec7)

Same result within Explorer with other labels
![image](https://github.com/user-attachments/assets/c9785e63-c998-4838-b8f2-1685f81639f7)

I used modify() to see the results, my aim is remove() or menu with vis=0 in case my bro need some features, I can put them back

**Comment by quanticplus04** [comment]
Nice help RubicBG. It will give an answer for many people. Dropbox entries are so envasive, my bro couldn't even see the whole list of his context menu !

**Comment by rudenko-ks** [comment]
A more complete version with the icon in the menu.
Thanks to everyone who wrote earlier in this topic - they helped a lot)

```
$path_dropbox_dll = reg.get('HKCR\CLSID\{ECD97DE5-3C8F-4ACB-AEEE-CCAB78F7711C}\InprocServer32')
menu(title = 'Dropbox' image = image.res(path_dropbox_dll, 0)) { }
modify(menu='Dropbox' where=this.title(
    str.res(path_dropbox_dll, -210),
    str.res(path_dropbox_dll, -212),
    str.res(path_dropbox_dll, -214),
    str.res(path_dropbox_dll, -216),
    str.res(path_dropbox_dll, -217),
    str.res(path_dropbox_dll, -218),
    str.res(path_dropbox_dll, -220),
    str.res(path_dropbox_dll, -221),
    str.res(path_dropbox_dll, -222),
    str.res(path_dropbox_dll, -223),
    str.res(path_dropbox_dll, -224),
    str.res(path_dropbox_dll, -225),
    str.res(path_dropbox_dll, -226),
    str.res(path_dropbox_dll, -227),
    str.res(path_dropbox_dll, -231),
    str.res(path_dropbox_dll, -232),
    str.res(path_dropbox_dll, -233),
    str.res(path_dropbox_dll, -239),
    str.res(path_dropbox_dll, -240),
    str.res(path_dropbox_dll, -241),
    str.res(path_dropbox_dll, -244),
    str.res(path_dropbox_dll, -247),
    str.res(path_dropbox_dll, -248),
    str.res(path_dropbox_dll, -249)
))
```

<details>
  <summary>View screenshot</summary>

  ![image](https://github.com/user-attachments/assets/6fa47a14-56d6-453e-b622-107345418fcf)

</details>

**Comment by RubicBG** [comment]
https://github.com/RubicBG/Nilesoft-Shell-Snippets/blob/main/ex2.user.cloud.share/cloud.dropbox.md

**Reply by rudenko-ks** [reply]
This link to the main page and half of the topics on the discussion forum would not have been created :)
Thx!
